### ISSUE TITLE
required header file missing

### ISSUE THREAD

**Issue by LukasvdWiel** [kokkos/kokkos#7157]
Dear Kokkos programmer,

source file: core/src/impl/Kokkos_MemoryPool.cpp

used the variable: type uint32_t

However, this type is not present in the recent C++ standard, such as used in GCC 14.1.0, giving rise to compile time errors such as:
```

/trinity/opt/apps/codes/Aspect/Trilinos/Trilinos-trilinos-release-13-2-0/packages/kokkos/core/src/impl/Kokkos_MemoryPool.cpp:112:48: error: ‘uint32_t’ has not been declared
  112 | void _print_memory_pool_state(std::ostream& s, uint32_t const* sb_state_ptr,
      |                                                ^~~~~~~~
/trinity/opt/apps/codes/Aspect/Trilinos/Trilinos-trilinos-release-13-2-0/packages/kokkos/core/src/impl/Kokkos_MemoryPool.cpp:113:49: error: ‘uint32_t’ has not been declared
  113 |                               int32_t sb_count, uint32_t sb_size_lg2,
      |                                                 ^~~~~~~~
/trinity/opt/apps/codes/Aspect/Trilinos/Trilinos-trilinos-release-13-2-0/packages/kokkos/core/src/impl/Kokkos_MemoryPool.cpp:114:31: error: ‘uint32_t’ has not been declared
  114 |                               uint32_t sb_state_size, uint32_t state_shift,
      |                               ^~~~~~~~
```
and a few more.

This can be remedied by following the suggestion by Selbie on

https://stackoverflow.com/questions/11069108/uint32-t-does-not-name-a-type

and adding:

`#include <stdint.h>`

to the top of that c source.
With that, libkokkos compiles again :-)

Have an excellent day,
Lukas

**Comment by dalg24** [comment]
Would you want to open a PR to resolve the issue?
(If so please include `<cstdint>` instead)

**Comment by dalg24** [comment]
Good point Daniel.  Closing since that is already resolved.
The fix should land in Trilinos shortly when we release 4.4

**Comment by masterleinad** [comment]
The fix has been part of 4.1.00 already (which is part of Trilinos 14.0.0).
